src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -933,10 +934,35 @@ FixedwingPositionControl::control_position(const math::Vector<2> &curr_pos, cons
 			_att_sp.yaw_body = 0;
 		}
 
+	} else if (_control_mode.flag_control_altitude_enabled
+		   && _control_mode.flag_control_offboard_enabled
+		   && pos_sp_curr.valid) {
+		/* Offboard altitude mode, control on yaw rate with fixed altitude */
+		_control_mode_current = FW_POSCTRL_MODE_OFFBOARD_ALTITUDE;
+
+		_hold_alt = _global_pos.alt;
+		float airspeed = _parameters.airspeed_trim;
+		float throttle = _parameters.throttle_cruise;
+		float gnd_speed = ground_speed.length();
+		tecs_update_pitch_throttle(pos_sp_curr.z,
+					   calculate_target_airspeed(airspeed),
+					   radians(_parameters.pitch_limit_min) - _parameters.pitchsp_offset_rad,
+					   radians(_parameters.pitch_limit_max) - _parameters.pitchsp_offset_rad,
+					   _parameters.throttle_min,
+					   _parameters.throttle_max,
+					   throttle,
+					   false,
+					   radians(_parameters.pitch_limit_min));
+
+		// Calculate the bank angle needed to achieve the yaw rate
+		float roll_sp = atan2f(pos_sp_curr.yawspeed * gnd_speed, 9.8065f);
```

Comment:
I'm not the most up to date on the offboard control flow, but is it meant here then that the offboard controller is commanding yaw rate via defining new position setpoints every timestep? This seems a bit strange. But I could just be out of the loop. Is there no other way to send rate setpoints from offboard?

Also - what happens when the ground speed is "negative"? If the intention is to have an offboard controller set the yaw rates needed to achieve some objective, maybe it is architecturally better to put acceleration mappings (e.g. as is done here with the level flight coordinated turn assumption to a roll set point) on the offboard controller then? And just send an attitude setpoint via offboard control? This way I guess it also wouldnt need to differ too much from the original altitude mode, just changing where the signals come from.

Reply:
As of right now there is no way to set a type mask to ignore specific parts of the attitude message (i.e use roll only). Therefore, the pitch correction will not be applied if both a `attitude_setpoint` as well as `alititude` is given to the controllers.